ui/src/app/machines/views/MachineDetails/MachineStorage/CacheSetsTable/CacheSetsTable.tsx

```diff
@@ -1,87 +1,54 @@
 import { MainTable } from "@canonical/react-components";
+import { useSelector } from "react-redux";
 
-import type { NormalisedStorageDevice as StorageDevice } from "../types";
-import { formatSize } from "../utils";
+import { formatSize, isCacheSet } from "../utils-new";
 
-import TableHeader from "app/base/components/TableHeader";
-import { useTableSort } from "app/base/hooks";
+import type { TSFixMe } from "app/base/types";
+import machineSelectors from "app/store/machine/selectors";
+import type { Machine } from "app/store/machine/types";
+import type { RootState } from "app/store/root/types";
 
-const getSortValue = (
-  sortKey: keyof StorageDevice,
-  storageDevice: StorageDevice
-) => storageDevice[sortKey];
+type Props = { systemId: Machine["system_id"] };
 
-type Props = { cacheSets: StorageDevice[] };
+const CacheSetsTable = ({ systemId }: Props): JSX.Element | null => {
+  const machine = useSelector((state: RootState) =>
+    machineSelectors.getById(state, systemId)
+  );
 
-const CacheSetsTable = ({ cacheSets }: Props): JSX.Element => {
-  const { currentSort, sortRows, updateSort } = useTableSort(getSortValue, {
-    key: "name",
-    direction: "descending",
-  });
-  // TODO: update useTableSort to TS with generics
-  // https://github.com/canonical-web-and-design/maas-ui/issues/1869
-  const sortedCacheSets = sortRows(cacheSets) as StorageDevice[];
+  if (machine && "disks" in machine) {
+    const rows: TSFixMe[] = [];
```

Comment:
Is setting the type to `TSFixMe[]` for strict mode? If so could you use a reduce function instead to generate the rows?

Reply:
Ah yeah I could! Thanks I'll make the change now

Reply:
So using a reduce function doesn't solve strict mode completely - gives me the error `Cannot assign type 'X' to type 'never'` for all of the columns content. I still think using reduce is neater, but I'm leaving `TSFixMe[]`in as the generic for now.